### PR TITLE
Issue #6272 Added Support Bootstrap Flutter on macOS

### PR DESCRIPTION
**Pull request by saymanrifat**
#6272

### CODE REVIEW

**Comment by sschuberth** [comment]
Thanks @saymanrifat for your contribution! Please have a look at the failing tests above. In particular, `git commit --amend -s` your commit to add a `Signed-off-by` footer to pass the DCO check. Also, please rewrite your commit message title to adhere to [Conventional Commits](https://www.conventionalcommits.org/).

**Comment by sschuberth** [comment]
Also, please move the reference to the issue to the commit message body by writing in there: `Fixes #6272.` (Note the trailing dot at the end of the sentence.)

File: plugins/package-managers/pub/src/main/kotlin/Pub.kt [hunk 1]
```diff
@@ -172,6 +172,14 @@ class Pub(
         val archive = when {
             Os.isWindows -> "windows/flutter_windows_$flutterVersion.zip"
             Os.isLinux -> "linux/flutter_linux_$flutterVersion.tar.xz"
+            Os.isMac -> {
+                val arch = when {
+                    System.getProperty("os.arch") == "x86_64" -> ""
+                    System.getProperty("os.arch") == "aarch64" -> "_arm64"
+                    else -> throw IllegalArgumentException("Unsupported architecture on macOS.")
+                }
```
Comment (sschuberth) [inline plugins/package-managers/pub/src/main/kotlin/Pub.kt:180]: Can be simplified to

                val arch = when (System.getProperty("os.arch")) {
                    "x86_64" -> ""
                    "aarch64" -> "_arm64"
                    else -> throw IllegalArgumentException("Unsupported architecture on macOS.")
                }

**Review by sschuberth — CHANGES_REQUESTED** [review]

File: plugins/package-managers/pub/src/main/kotlin/Pub.kt [hunk 2]
```diff
@@ -172,6 +172,14 @@ class Pub(
         val archive = when {
             Os.isWindows -> "windows/flutter_windows_$flutterVersion.zip"
             Os.isLinux -> "linux/flutter_linux_$flutterVersion.tar.xz"
+            Os.isMac -> {
+                val arch = when {
+                    System.getProperty("os.arch") == "x86_64" -> ""
+                    System.getProperty("os.arch") == "aarch64" -> "_arm64"
+                    else -> throw IllegalArgumentException("Unsupported architecture on macOS.")
```
Comment (fviernau) [inline plugins/package-managers/pub/src/main/kotlin/Pub.kt:179]: Would it make sense to include '$arch' into the exception message?

**Review by fviernau — CHANGES_REQUESTED** [review]

File: plugins/package-managers/pub/src/main/kotlin/Pub.kt (hunk 2 above)
Comment (saymanrifat) [reply plugins/package-managers/pub/src/main/kotlin/Pub.kt:179]: Yep, I think so. I will

**Review by saymanrifat — COMMENTED** [review]
Fixes #6272

**Pull request closed without merging**